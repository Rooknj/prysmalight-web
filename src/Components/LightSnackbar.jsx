import React from "react";
import Snackbar from "@material-ui/core/Snackbar";
import IconButton from "@material-ui/core/IconButton";
import CloseIcon from "@material-ui/icons/Close";
import styled from "styled-components";
import Typography from "@material-ui/core/Typography";

const StyledIconButton = styled(IconButton)`
  padding: ${({ theme }) => theme.spacing.unit / 4}px;
`;

const propTypes = {};
const defaultProps = {};

class LightSnackbar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      open: true
    };
  }
  handleClose = (event, reason) => {
    if (reason === "clickaway") {
      return;
    }
    this.setState({ open: false });
  };
  render() {
    const { message, type } = this.props;
    return (
      <React.Fragment>
        <Snackbar
          anchorOrigin={{
            vertical: "bottom",
            horizontal: "center"
          }}
          open={this.state.open}
          autoHideDuration={6000}
          onClose={this.handleClose}
          ContentProps={{
            "aria-describedby": "message-id"
          }}
          message={
            <span id="message-id">
              <Typography variant="body1" color={type === "error" ? "error" : "inherit"}>
                {message}
              </Typography>
            </span>
          }
          action={[
            <StyledIconButton
              key="close"
              aria-label="Close"
              color="inherit"
              onClick={this.handleClose}
            >
              <CloseIcon />
            </StyledIconButton>
          ]}
        />
      </React.Fragment>
    );
  }
}
LightSnackbar.propTypes = propTypes;
LightSnackbar.defaultProps = defaultProps;
export default LightSnackbar;
