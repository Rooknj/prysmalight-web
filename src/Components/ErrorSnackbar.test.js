import React from "react";
import { shallow, mount } from "enzyme";
import { createSerializer } from "enzyme-to-json";
import LightSnackbar from "./LightSnackbar";
import { ThemeProvider } from "styled-components";

// Add serializer to allow enzyme to create snapshots
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

const mockTheme = { spacing: { unit: 2 } };
const mockMessage = "Error!";
// TODO: Test rendering with actual mock lights
it("renders without crashing", () => {
  shallow(
    <ThemeProvider theme={mockTheme}>
      <LightSnackbar message={mockMessage} />
    </ThemeProvider>
  );
});

it("matches shallow snapshot", () => {
  const shallowed = shallow(
    <ThemeProvider theme={mockTheme}>
      <LightSnackbar message={mockMessage} />
    </ThemeProvider>
  );
  expect(shallowed).toMatchSnapshot();
});

it("matches mounted snapshot", () => {
  const mounted = mount(
    <ThemeProvider theme={mockTheme}>
      <LightSnackbar message={mockMessage} />
    </ThemeProvider>
  );
  expect(mounted).toMatchSnapshot();
});
