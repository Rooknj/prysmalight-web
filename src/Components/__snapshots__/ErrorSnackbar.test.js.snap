// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches mounted snapshot 1`] = `
Array [
  <div
    className="MuiSnackbar-root-1 MuiSnackbar-anchorOriginBottomCenter-3"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      aria-describedby="message-id"
      className="MuiTypography-root-38 MuiTypography-body1-47 MuiPaper-root-11 MuiPaper-elevation6-19 MuiSnackbarContent-root-8"
      role="alertdialog"
      style={Object {}}
    >
      <div
        className="MuiSnackbarContent-message-9"
      >
        <span
          id="message-id"
        >
          <p
            className="MuiTypography-root-38 MuiTypography-body1-47 MuiTypography-colorInherit-67"
          >
            Error!
          </p>
        </span>
      </div>
      <div
        className="MuiSnackbarContent-action-10"
      >
        <Styled(WithStyles(IconButton))
          aria-label="Close"
          color="inherit"
          key="close"
          onClick={[Function]}
        >
          <button
            aria-label="Close"
            className="MuiButtonBase-root-80 MuiIconButton-root-74 MuiIconButton-colorInherit-75 sc-bdVaJa dKqnEJ"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-79"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-83"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-92"
            />
          </button>
        </Styled(WithStyles(IconButton))>
      </div>
    </div>
  </div>,
  "",
]
`;

exports[`matches shallow snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
